src/scenechange/mod.rs

```diff
@@ -104,237 +140,288 @@ impl SceneChangeDetector {
-    // to enable early loop exit if we find a scene flash.
-    for j in (1..=lookahead_distance).rev() {
-      let result = self.has_scenecut(
-        frame_subset[0].clone(),
-        frame_subset[j].clone(),
-        frameno - 1 + j as u64,
+    scenecut
+  }
+
+  // Initially fill score deque with frame scores
+  fn initialize_score_deque(
+    &mut self, frame_set: &[Arc<Frame<T>>], input_frameno: u64,
+    previous_keyframe: u64, init_len: usize,
+  ) {
+    for x in 0..init_len {
+      self.run_comparison(
+        frame_set[x].clone(),
+        frame_set[x + 1].clone(),
+        input_frameno,
         previous_keyframe,
       );
-      debug!(
-        "[SF-Detect-1] Frame {} to {}: I={:.3} T={:.3} P={:.3} {}",
-        frameno - 1,
-        frameno - 1 + j as u64,
-        result.intra_cost,
-        result.threshold,
-        result.inter_cost,
-        if result.has_scenecut { "No flash" } else { "Scene flash" }
-      );
-      if !result.has_scenecut {
-        // Any frame in between `0` and `j` cannot be a real scenecut.
-        for i in 0..=j {
-          let frameno = frameno + i as u64 - 1;
-          self.excluded_frames.insert(frameno);
-        }
-        // Because all frames in this gap are already excluded,
-        // exit the loop early as an optimization.
-        break;
-      }
     }
+  }
 
-    // Where A-F are scenes: AAAAABBCCDDEEFFFFFF
-    // If each of BB ... EE are shorter than `lookahead_distance`, they are
-    // detected as flashes and not considered scenecuts.
-    // Instead, the first F frame becomes a scenecut.
-    // If the video ends before F, no frame becomes a scenecut.
-    for i in 1..lookahead_distance {
-      let result = self.has_scenecut(
-        frame_subset[i].clone(),
-        frame_subset[lookahead_distance].clone(),
-        frameno - 1 + lookahead_distance as u64,
-        previous_keyframe,
-      );
-      debug!(
-        "[SF-Detect-2] Frame {} to {}: I={:.3} T={:.3} P={:.3} {}",
-        frameno - 1 + i as u64,
-        frameno - 1 + lookahead_distance as u64,
-        result.intra_cost,
-        result.threshold,
-        result.inter_cost,
-        if result.has_scenecut { "Scene flash" } else { "No flash" }
-      );
-      if result.has_scenecut {
-        // If the current frame is the frame before a scenecut, it cannot also be the frame of a scenecut.
-        let frameno = frameno + i as u64 - 1;
-        self.excluded_frames.insert(frameno);
+  /// Runs scene change comparison beetween 2 given frames
+  /// Insert result to start of score deque
+  fn run_comparison(
+    &mut self, frame1: Arc<Frame<T>>, frame2: Arc<Frame<T>>,
+    input_frameno: u64, previous_keyframe: u64,
+  ) {
+    let result = if self.fast_mode {
+      self.fast_scenecut(frame1, frame2)
+    } else {
+      self.cost_scenecut(frame1, frame2, input_frameno, previous_keyframe)
+    };
+    self
+      .score_deque
+      .insert(0, (result.inter_cost as f64, result.intra_cost as f64));
```

Comment:
The threshold computation in `cost_scenecut()` is thrown out here.
```suggestion
      .insert(0, (result.inter_cost as f64, result.threshold as f64));
```

Reply:
Fixed